lib/util/cache/repository/index.spec.ts

```diff
@@ -38,6 +38,6 @@ describe('util/cache/repository/index', () => {
     await initRepoCache({ ...config, repositoryCache: 'reset' });
     expect(fs.readCacheFile).not.toHaveBeenCalled();
     expect(fs.outputCacheFile).toHaveBeenCalled();
-    expect(getCache()).toBeEmpty();
+    expect(getCache()).toEqual({ modified: true });
```

Comment:
To be honest I dont like this being part of the cache data to begin with.

What about adding additional function to the `RepoCache`
i.e.

```ts
export interface RepoCache {
  load(): Promise<void>;
  save(): Promise<void>;
  getData(): RepoCacheData;
  isModified(): boolean | undefined;
}
```
`isModified` will basically compare the current cache hash with the `oldCacheHash`. this way no other modifications are needed to other types.

This will return `boolean` according to this comparison, or undefined for when comparison cant be made i.e. no `oldHash`. 

@zharinov @viceice wdyt?

Reply:
sounds good